spec/update-process-env-spec.js

```diff
@@ -84,5 +84,33 @@ describe('updateProcessEnv(launchEnv)', function () {
         expect(process.env).toEqual({FOO: 'bar'})
       })
     })
+
+    describe('shells on osx', function () {
+      it('envShouldBePatched() returns the shell when the shell should be patched', function () {
```

Comment:
The `envShouldBePatched` name is a little clearer, but I think it'd be best to make it clear that the return value depends on the shell, not the env. Also, the env is completely replaced (except for the 3 'special' variables), so emphasizing the 'patching' doesn't seem entirely necessary.

How about `shouldGetEnvFromShell`?